### PR TITLE
 Move 'deadline', 'expires', and 'created' from the hook definition to the task template and use triggerHook method to trigger the hook

### LINKED ISSUE #915: `deadline` and `expires` fields are deprecated and will soon be removed
We should fix our projects to stop using these fields in hook definitions. They are going to be moved to the `task` template: https://docs.taskcluster.net/reference/core/taskcluster-hooks/docs/firing-hooks#task-times.

See https://github.com/taskcluster/taskcluster-hooks/pull/85.

@garbas @La0 I think we will need a new release for this.

### PR DESCRIPTION
**Pull request by marco-c**
Fixes #915.

r? @La0 for the shipit_pulse_listener parts
r? @garbas for the Nix stuff (I bet there's a cleaner way to mkTaskclusterMergeEnv)

### CODE REVIEW

File: nix/lib/default.nix [hunk 1]
```diff
@@ -140,6 +153,9 @@ in rec {
            tags workerType;
         payload = mkTaskclusterTaskPayload payload;
         metadata = mkTaskclusterTaskMetadata metadata;
+        created = { "$fromNow" = created; };
+        deadline = { "$fromNow" = deadline; };
+        expires = { "$fromNow" = expires; };
```
Comment (La0) [inline nix/lib/default.nix]: This syntax is not supported by the taskcluster lib we use.

**Comment by marco-c** [comment]
It doesn't work:
> 2018-03-12T16:28:51.390440+00:00 app[worker.1]: taskcluster.exceptions.TaskclusterRestFailure: data should NOT have additional properties

**Comment by marco-c** [comment]
Let's see if we can figure it out, or we have to revert this.

**Comment by marco-c** [comment]
We need to update the triggerSchema too, as the payload to triggerHook must match the schema.

**Pull request merged**